logzio/resource_drop_filter.go

```diff
@@ -0,0 +1,279 @@
+		Read:   resourceDropFilterRead,
+		Update: resourceDropFilterUpdate,
+		Delete: resourceDropFilterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			dropFilterIdField: {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			dropFilterActive: {
+				Type:     schema.TypeBool,
+				Computed: true,
+				Optional: true,
+			},
+			dropFilterLogType: {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			dropFilterFieldConditions: {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						dropFilterFieldName: {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						dropFilterValue: {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Second),
+			Update: schema.DefaultTimeout(5 * time.Second),
+			Delete: schema.DefaultTimeout(5 * time.Second),
+		},
+	}
+}
+
+// Creates a new drop filter in logzio
+func resourceDropFilterCreate(d *schema.ResourceData, m interface{}) error {
+	createDropFilter := createCreatDropFilterFromSchema(d)
+	active, isSet := d.GetOkExists(dropFilterActive)
+	if isSet {
+		log.Printf("active attribute is set to %t, note that this field is ignored for creation. A drop filter will always be active after creation.\n", active)
+	}
+	dropFilter, err := dropFilterClient(m).CreateDropFilter(createDropFilter)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(dropFilter.Id)
+
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		err = resourceDropFilterRead(d, m)
+		if err != nil {
+			if strings.Contains(err.Error(), "could not find drop filter with id") {
+				return resource.RetryableError(err)
+			}
+		}
+
+		return resource.NonRetryableError(err)
+	})
+}
+
+// Gets drop filter by id
+func resourceDropFilterRead(d *schema.ResourceData, m interface{}) error {
+	dropFilters, err := dropFilterClient(m).RetrieveDropFilters()
+	if err != nil {
+		return err
+	}
+
+	dropFilter := findDropFilterById(d.Id(), dropFilters)
+	if dropFilter == nil {
+		return fmt.Errorf("could not find drop filter with id: %s", d.Id())
+
+	}
+
+	setDropFilter(d, dropFilter)
+	return nil
+}
+
+// Updates drop field by id - activate or deactivate
+func resourceDropFilterUpdate(d *schema.ResourceData, m interface{}) error {
+	activate := d.Get(dropFilterActive).(bool)
+	var err error
+	if activate {
+		_, err = dropFilterClient(m).ActivateDropFilter(d.Id())
+	} else {
+		_, err = dropFilterClient(m).DeactivateDropFilter(d.Id())
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
```

Comment:
didnt understand why we need this

Reply:
A retry? I think it's best practice to add a retry for every resource...